src/Serilog/Sinks/PeriodicBatching/PeriodicBatchingSink.cs

```diff
@@ -41,19 +41,21 @@ namespace Serilog.Sinks.PeriodicBatching
     /// </remarks>
     public abstract class PeriodicBatchingSink : ILogEventSink, IDisposable
     {
+        const int NotStarted = 0;
+        const int Started = 1;
+        const int Unloading = 2;
+
         readonly int _batchSizeLimit;
         readonly ConcurrentQueue<LogEvent> _queue;
         readonly BatchedConnectionStatus _status;
         readonly Queue<LogEvent> _waitingBatch = new Queue<LogEvent>();
 
-        readonly object _stateLock = new object();
 #if NO_TIMER
         readonly PortableTimer _timer;
 #else
         readonly Timer _timer;
 #endif
-        bool _unloading;
-        bool _started;
+        volatile int _state;
```

Comment:
I think using `Interlocked` along with `volatile` will generate a compiler warning (IIRC). Do we get one in the usages below? If so and we're happy it is benign we should `#pragma` it out.